tools/release/release-output/output-validations.ts

```diff
@@ -27,6 +28,46 @@ export function checkReleaseBundle(bundlePath: string): string[] {
   return failures;
 }
 
+/**
+ * Checks the specified TypeScript definition file by ensuring it does not contain invalid
+ * dynamic import statements. There can be invalid type imports paths because we compose the
+ * release package by moving things in a desired output structure. See Angular package format
+ * specification and https://github.com/angular/material2/pull/12876
+ */
+export function checkTypeDefinitionFile(filePath: string): string[] {
+  const baseDir = dirname(filePath);
+  const fileContent = readFileSync(filePath, 'utf8');
+  const failures = [];
+
+  const sourceFile = ts.createSourceFile(filePath, fileContent, ts.ScriptTarget.Latest, true);
+  const nodeQueue = [...sourceFile.getChildren()];
+
+  while (nodeQueue.length) {
```

Comment:
Rather than building up the queue yourself, you can use `SourceFile.forEachChild` which goes through all the nodes recursively.

Reply:
I'm aware of that utility function, but don't really see any big value in it since we would need to call `forEachChild` for all child nodes again.

Additionally for future checks we could just check in breadth before going into depth. Not really desired yet though.

Reply:
That was my point: if I remember correctly, it’s recursive so we don’t have to call it for the children. You can see how we’re using it in the breaking changes utils.

Reply:
Based on my understanding and the TypeScript Wiki, it isn't recursive.

Where can you see it in the breaking changes stuff?

Reply:
Looks like I was remembering it wrong and `forEachChild` isn't recursive. The breaking change utils are using `forEachComment` from `tsutils`.